workitem.2.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+	query "github.com/almighty/almighty-core/query/simple"
+	"github.com/almighty/almighty-core/transaction"
+	"github.com/goadesign/goa"
+)
+
+// Workitem2Controller implements the workitem.2 resource.
+type Workitem2Controller struct {
+	*goa.Controller
+	wiRepository models.WorkItemRepository
+	ts           transaction.Support
+}
+
+// NewWorkitem2Controller creates a workitem.2 controller.
+func NewWorkitem2Controller(service *goa.Service, wiRepository models.WorkItemRepository, ts transaction.Support) *Workitem2Controller {
+	return &Workitem2Controller{Controller: service.NewController("WorkitemController"), wiRepository: wiRepository, ts: ts}
+}
+
+// List runs the list action.
+// Prev and Next links will be present only when there actually IS a next or previous page.
+// Last will always be present. Total Item count needs to be computed from the "Last" link.
+func (c *Workitem2Controller) List(ctx *app.ListWorkitem2Context) error {
+	// Workitem2Controller_List: start_implement
+
+	exp, err := query.Parse(ctx.Filter)
+	if err != nil {
+		return ctx.BadRequest(goa.ErrBadRequest(fmt.Sprintf("could not parse filter: %s", err.Error())))
+	}
+	var offset int
+	var limit int
+
+	if ctx.PageOffset == nil {
+		offset = 0
+	} else {
+		offset = int(*ctx.PageOffset)
+	}
+	if err != nil {
+		return ctx.BadRequest(goa.ErrBadRequest(fmt.Sprintf("could not parse page offset: %s", err.Error())))
+	}
+
+	if ctx.PageLimit == nil {
+		limit = 100
+	} else {
+		limit = int(*ctx.PageLimit)
+	}
+	if err != nil {
+		return ctx.BadRequest(goa.ErrBadRequest(fmt.Sprintf("could not parse page limit: %s", err.Error())))
+	}
+	if offset < 0 {
+		return ctx.BadRequest(goa.ErrBadRequest(fmt.Sprintf("offset must be >= 0, but is: %d", offset)))
```

Comment:
My thinking was that this is a programming error, something we want to catch instead of silently ignoring it. What scenario do you have in mind?